scripts/install-calico-windows.ps1

```diff
@@ -0,0 +1,227 @@
+        [parameter(Mandatory=$true)] $NewString
+    )
+
+    (Get-Content c:\CalicoWindows\config.ps1).replace($OldString, $NewString) | Set-Content c:\CalicoWindows\config.ps1 -Force
+}
+
+function StartCalico()
+{
+    Write-Host "`nStart Calico...`n"
+
+    pushd
+    cd c:\CalicoWindows
+    .\install-calico.ps1
+    popd
+    Write-Host "`nCalico Started`n"
+}
+
+$Backend="vxlan"
+$BaseDir="c:\k"
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+
+$helper = "$BaseDir\helper.psm1"
+$helperv2 = "$BaseDir\helper.v2.psm1"
+md $BaseDir -ErrorAction Ignore
+if (!(Test-Path $helper))
+{
+    Invoke-WebRequest https://raw.githubusercontent.com/Microsoft/SDN/master/Kubernetes/windows/helper.psm1 -O $BaseDir\helper.psm1
+}
+if (!(Test-Path $helperv2))
+{
+    Invoke-WebRequest https://raw.githubusercontent.com/Microsoft/SDN/master/Kubernetes/windows/helper.v2.psm1 -O $BaseDir\helper.v2.psm1
+}
+ipmo -force $helper
+ipmo -force $helperv2
+
+$platform=GetPlatformType
+
+if (-Not [string]::IsNullOrEmpty($KubeVersion) -and $platform -NE "eks") {
+    PrepareKubernetes
+}
+
+Write-Host "Download Calico for Windows release..."
+DownloadFile -Url $ReleaseBaseURL/$ReleaseFile -Destination c:\calico-windows.zip
+Write-Host "Unzip Calico for Windows release..."
+Expand-Archive c:\calico-windows.zip c:\
+
+Write-Host "Setup Calico for Windows..."
+SetConfigParameters -OldString '<your datastore type>' -NewString $Datastore
+SetConfigParameters -OldString '<your etcd endpoints>' -NewString "$EtcdEndpoints"
+SetConfigParameters -OldString '<your service cidr>' -NewString $ServiceCidr
+SetConfigParameters -OldString '<your dns server ips>' -NewString $DNSServerIPs
+SetConfigParameters -OldString 'KUBECONFIG = "c:\k\config"' -NewString 'KUBECONFIG = "c:\CalicoWindows\calico-kube-config"'
+
+if ($platform -EQ "azure") {
+    Write-Host "Setup Calico for Windows for azure..."
+    $Backend="none"
+    SetConfigParameters -OldString 'CALICO_NETWORKING_BACKEND="vxlan"' -NewString 'CALICO_NETWORKING_BACKEND="none"'
+    SetConfigParameters -OldString 'KUBE_NETWORK = "Calico.*"' -NewString 'KUBE_NETWORK = "azure.*"'
+    GetCalicoKubeConfig -SecretName 'calico-windows'
+}
+if ($platform -EQ "eks") {
+    $awsNodeName = Invoke-RestMethod -uri http://169.254.169.254/latest/meta-data/local-hostname -ErrorAction Ignore
+    Write-Host "Setup Calico for Windows for eks, node name $awsNodeName ..."
+    $Backend="none"
+    $awsNodeNameQuote = """$awsNodeName"""
+    SetConfigParameters -OldString '$(hostname).ToLower()' -NewString "$awsNodeNameQuote"
+    SetConfigParameters -OldString 'CALICO_NETWORKING_BACKEND="vxlan"' -NewString 'CALICO_NETWORKING_BACKEND="none"'
+    SetConfigParameters -OldString 'KUBE_NETWORK = "Calico.*"' -NewString 'KUBE_NETWORK = "vpc.*"'
+
+    # Note: EKS mode requires downloading kubectl.exe to c:\k before running this script.
+    # https://docs.aws.amazon.com/eks/latest/userguide/install-kubectl.html
```

Comment:
@lmm I think we should ask user to download kubectl.exe and document Calico on EKS as the first step. The instructions already provided by aws. Downloading whole kubernetes binaries seems too expensive. WDYT?

Reply:
@song-jiang yeah I was wondering about this too. I think it's a good idea anyways since it stops the user from potentially running kubelet or kube-proxy from our scripts, which might be a bad idea since those are managed by AWS services on the node. I'll update this.

Reply:
Fixed